### PR TITLE
chore(core): Update dependencies 15.07.2022

### PR DESCRIPTION
**Pull request by JoseLion**
This PR updates all the dependencies of the library and the example project. the main benefit of this upgrade is to provide compatibility with React v18 and to ensure the library works with the latest version of React Native and `react-native-svg`. Some notes to take into account:
- Switched the project to Yarn v3 in `nodeLinker: node-modules` mode
- The example project now uses Hermes Engine
- The changes on the native project (`example/ios/` and `example/android/` folders) are based on the [React Native Upgrade Helper tool](https://react-native-community.github.io/upgrade-helper/?from=0.66.4&to=0.69.3)
- Removed dependencies and configurations that are no longer necessary (like `react-native-typescript-transformer`)
- In the example, moved the application files to the `src/` folder
- Added a `dev` script to simplify the process of testing the library into the example application
  - Running `yarn dev` will build and install the local version of the library in the example app
- Fixed a few typing issues in the library
- Split TS configuration to have dedicated build/production settings
- **Important:** Native driver animations no longer work in iOS either
  - The upgrade to React Native showed a problem with how the animations are triggered
  - We need to refactor the animation flow to make it async and allow waiting for other animations while using the native driver
  - We should also allow users to opt-in/opt-out of using the native driver to allow animations troubleshooting

> **PS:** CI build time was reduced from ~1m40s to ~48s thanks to the new configuration!

### CODE REVIEW

**Review by kdquistanchala — APPROVED** [review]

File: example/android/app/src/debug/java/com/spotlightexample/ReactNativeFlipper.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /**
- * Copyright (c) Facebook, Inc. and its affiliates.
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
```
Comment (Alex0jk) [inline example/android/app/src/debug/java/com/spotlightexample/ReactNativeFlipper.java:2]: Branding update! haha

File: example/src/App.tsx [hunk 2]
```diff
@@ -87,7 +87,7 @@ export const App: React.FC = () => {
           bounciness: 100,
           speed: 1,
           toValue: Dimensions.get("screen").height * 0.25,
-          useNativeDriver: Platform.OS !== "android"
+          useNativeDriver: false
```
Comment (Alex0jk) [inline example/src/App.tsx:90]: I wonder, how will this impact performance on iOS device, and also how we can address any issues if the impact is meaningful

File: src/lib/tour-overlay/TourOverlay.component.tsx [hunk 3]
```diff
@@ -48,14 +49,14 @@ export const TourOverlay = React.forwardRef<TourOverlayRef, TourOverlayProps>((p
   const useNativeDriver = useMemo(() => Platform.select({
     android: false,
     default: false,
-    ios: true
+    ios: false
```
Comment (Alex0jk) [inline src/lib/tour-overlay/TourOverlay.component.tsx:52]: Is is needed to specify the Native Driver for each OS if all of them are false?

**Review by Alex0jk — APPROVED** [review]
I looks good, I left a few comments about the native driver stuff to talk about.
I do wonder why the animation are not working with the native driver on iOS, and if it is something of a high priority to overcome (I did see that we need to refactor our animation flow).

File: src/lib/tour-overlay/TourOverlay.component.tsx (hunk 3 above)
Comment (Alex0jk) [reply src/lib/tour-overlay/TourOverlay.component.tsx:52]: I suppose that it is ok to have it like this if we want to then give a have to opt-in/opt-out. A comment would good to have explaining the change and the steps taken to support the Native Driver.

File: example/android/app/src/debug/java/com/spotlightexample/ReactNativeFlipper.java (hunk 1 above)
Comment (JoseLion) [reply example/android/app/src/debug/java/com/spotlightexample/ReactNativeFlipper.java:2]: Yeah 😂 this comes from the [React Native Upgrade Helper tool](https://react-native-community.github.io/upgrade-helper/?from=0.66.4&to=0.69.3), which will make future updates easier too 😁

File: example/src/App.tsx (hunk 2 above)
Comment (JoseLion) [reply example/src/App.tsx:90]: Yeah, it seems things have changed with animations in new React Native versions so we need to revisit the animation workflow. This is just the example code but I think it's a good idea to leave the comment until this is solved 🙂

File: src/lib/tour-overlay/TourOverlay.component.tsx (hunk 3 above)
Comment (JoseLion) [reply src/lib/tour-overlay/TourOverlay.component.tsx:52]: Yeah, for now it's false on both, but later users will be able to choose for themselves. I'll leave a comment here too. Thanks! 🙂

**Pull request merged**